data/processors/maps.py

```diff
@@ -124,112 +130,187 @@ def assign_roomfinder_maps(data):
-                    "width":  m["width"],
-                    "height": m["height"],
-                    "source": m.get("source", "Roomfinder"),
-                    "path":   m.get("path", f"webp/{m['id']}.webp")
-                }
-                for m in available_maps
-            ],
-        }
+        _save_map_data(available_maps, entry)
+
+
+def _save_map_data(available_maps, entry):
+    # Select map with lowest scale as default
+    default_map = available_maps[0]
+    for m in available_maps:
+        if int(m["scale"]) < int(default_map["scale"]):
+            default_map = m
+
+    roomfinder_map_data = {
+        "default": default_map["id"],
+        "available": [
+            {
+                "id": m["id"],
+                "scale": m["scale"],
+                "name": m["desc"],
+                "width": m["width"],
+                "height": m["height"],
+                "source": m.get("source", "Roomfinder"),
+                "path": m.get("path", f"webp/{m['id']}.webp")
+            }
+            for m in available_maps
+        ],
+    }
+    entry.setdefault("maps", {})["roomfinder"] = roomfinder_map_data
+
+
+def _set_maps_from_parent(data, entry):
+    building_parent = [e for e in entry["parents"] if data[e]["type"] == "building"]
+    # Verification of this already done for coords, see above
+    building_parent = data[building_parent[0]]
+    if not building_parent.get("maps", {}) \
+            .get("roomfinder", {}) \
+            .get("available", []):
+        roomfinder_map_data = {"is_only_building": True}
+        # Both share the reference now, assuming that the parening_parent
+        # will be processed some time later in this loop.
+        building_parent.setdefault("maps", {})["roomfinder"] = roomfinder_map_data
         entry.setdefault("maps", {})["roomfinder"] = roomfinder_map_data
-    
+    else:
+        # TODO: 5510.02.001
+        roomfinder_map_data = entry.setdefault("maps", {}).get("roomfinder", {})
+        roomfinder_map_data.update(building_parent["maps"]["roomfinder"])
+        roomfinder_map_data["is_only_building"] = True
+
+
+def _extract_available_maps(entry, custom_maps, maps_list):
+    available_maps = []
+    for (b_id, floor), m in custom_maps.items():
+        if entry["type"] == "room" and b_id in entry["parents"] and \
+                "tumonline_data" in entry and f".{floor}." in entry["tumonline_data"]["roomcode"]:
+            available_maps.append(m)
+    lat_coord, lon_coord = (entry["coords"]["lat"], entry["coords"]["lon"])
+    for m in maps_list:
+        # Assuming coordinates in Central Europe
+        if m["latlonbox"]["south"] < lat_coord < m["latlonbox"]["north"] and \
+                m["latlonbox"]["west"] < lon_coord < m["latlonbox"]["east"]:
+            available_maps.append(m)
+    return available_maps
+
+
+def _merge_str(s1: str, s2: str):
+    """Merges two strings. The Result is of the format common_prefix s1/s2 common_suffix"""
```

Comment:
```suggestion
    """
    Merges two strings. The Result is of the format common_prefix s1/s2 common_suffix.
    Example: "Thierschbau 5. OG" and "Thierschbau 6. OG" -> "Thierschbau 5/6. OG"
    """
```
This is quite fancy, but wasn't directly clear to me.